sched/wqueue/kwork_thread.c

```diff
@@ -0,0 +1,297 @@
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <unistd.h>
+#include <sched.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <assert.h>
+#include <queue.h>
+#include <debug.h>
+
+#include <nuttx/wqueue.h>
+#include <nuttx/kthread.h>
+#include <nuttx/semaphore.h>
+
+#include "wqueue/wqueue.h"
+
+#if defined(CONFIG_SCHED_WORKQUEUE)
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if defined(CONFIG_SCHED_CRITMONITOR_MAXTIME_WQUEUE) && CONFIG_SCHED_CRITMONITOR_MAXTIME_WQUEUE > 0
+#  define CALL_WORKER(worker, arg) \
+     do \
+       { \
+         uint32_t start; \
+         uint32_t elapsed; \
+         start = up_critmon_gettime(); \
+         worker(arg); \
+         elapsed = up_critmon_gettime() - start; \
+         if (elapsed > CONFIG_SCHED_CRITMONITOR_MAXTIME_WQUEUE) \
+           { \
+             serr("WORKER %p execute too long %"PRIu32"\n", \
+                   worker, elapsed); \
+           } \
+       } \
+     while (0)
+#else
+#  define CALL_WORKER(worker, arg) worker(arg)
+#endif
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+#if defined(CONFIG_SCHED_HPWORK)
+/* The state of the kernel mode, high priority work queue(s). */
+
+struct hp_wqueue_s g_hpwork;
+#endif /* CONFIG_SCHED_HPWORK */
+
+#if defined(CONFIG_SCHED_LPWORK)
+/* The state of the kernel mode, low priority work queue(s). */
+
+struct lp_wqueue_s g_lpwork;
+#endif /* CONFIG_SCHED_LPWORK */
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: work_thread
+ *
+ * Description:
+ *   These are the worker threads that performs the actions placed on the
+ *   high priority work queue.
+ *
+ *   These, along with the lower priority worker thread(s) are the kernel
+ *   mode work queues (also build in the flat build).
+ *
+ *   All kernel mode worker threads are started by the OS during normal
+ *   bring up.  This entry point is referenced by OS internally and should
+ *   not be accessed by application logic.
+ *
+ * Input Parameters:
+ *   argc, argv (not used)
+ *
+ * Returned Value:
+ *   Does not return
+ *
+ ****************************************************************************/
+
+static int work_thread(int argc, FAR char *argv[])
```

Comment:
```suggestion
static noreturn_function int work_thread(int argc, FAR char *argv[])
```
Since it does not return, should we add the `noreturn` attribute?

Reply:
Done. Thank you.

Reply:
I don't know if you've missed this change or if it has been reverted somehow, but `noreturn` attribute is missing.